docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc

```diff
@@ -0,0 +1,145 @@
+
+To distribute the load this compilation generates and to follow a decentralized approach, these jobs are run on the same cluster as they compile the catalog for.
+The Steward component on a cluster starts a compilation job if the `cluster.syn.tools/needs-compilation` annotation is set to `true` on its own cluster object.
+It will then launch the respective job to compile the cluster catalog.
+
+For clusters where this isn't possible due to resource, security or other constraints, a central system runs the jobs.
+These clusters need to be configured in the central system so it will react on the mentioned annotation an create the compile jobs for them.
+
+
+=== User Stories
+
+==== Config Change in Global Config Repo
+
+If configuration in the global inventory repo changes, Quartermaster needs to find all clusters which are affected by the changed configuration.
+
+
+==== Config Change in Tenant Config Repo
+
+If configuration in a tenant inventory repo changes, Quartermaster needs to find all clusters which are affected by the changed configuration.
+The set of clusters to consider are the clusters of the respective tenant.
+
+
+==== Change of Cluster Facts
+
+If the facts of a cluster change the set of clusters to consider already only consists of this cluster.
+Triggering the webhook needs to be implemented in Lieutenant (operator or API).
+
+
+==== New Release of a Commodore Component
+
+The new release of a Commodore component won't trigger anything directly.
+If the specific version for a component is being updated it results in a config change in either the global inventory repo or a tenant repo.
+This in turn results in the recompilation of the affected clusters.
+
+
+==== Manually Trigger a Compile
+
+To manually trigger a compile for developping or debugging purposes the webhook can be called with a tool like `curl`.
+Providing a cluster or tenant ID can limit the set of clusters to compile.
+
+
+=== Risks and Mitigations
+
+==== Rate Limiting
+
+Since the compilation is a relatively expensive operation resource wise, it might become necessary to implement some kind of rate limiting.
+For example could Quartermaster make sure that a certain amount of compile jobs per amount of time isn't exceeded for a single cluster or tenant.
+This would mitigate the situation where a lot of changes in various Git repos would result in many compile jobs which could exhaust the resources of a cluster.
+The downside would be a longer reaction time for changes to be applied on a cluster.
+
+
+== Alternatives
+
+An alternative would be to couple the two mentioned parts of the design together.
+This would mean that the same component is responsible to trigger new compiles and run the compiles.
+Such a solution would probably require a centralized
```

Comment:
this sentence looks incomplete, probably should be "require a centralized job runner" or something along those lines?